### ISSUE TITLE
Fix server->client initialization messages getting dropped

### ISSUE THREAD

**Issue by applmak** [google/chicago-brick#58]
Messages sent from server via network are dropped if clients haven't yet opened a connection on the same namespace. This makes initialization things (that one would normally put in willBeShownSoon) super annoying. Fix this by queuing such messages until the client actually connects.

**Comment by applmak** [comment]
This is even weirder than I thought: sometimes we fails to connect between the client & the server. After EXTENSIVE digging, I figured out that this is happening due to our abuse of the namespace protocol of socket.io. The namespace criteria itself is undocumented; I deduced that it couldn't have certain special characters like ',' in them, and constructed a namespace that I thought looked reasonable (e.g. '/module0X0X1920X1080X0', which is '/module' + rect + module deadline). In order to avoid a second round-trip, we actually connect to a namespace with additional data tacked on like a URL (e.g. /module0X0X1920X1080X0?id=0,0,1920,1080-0&rect=0,0,1920,1080), which seemed to work just fine in my testing. However, the actual code that parses a response on the server from the client looks like this:

```
p.nsp = '';
while (++i) {
  var c = str.charAt(i);
  if (',' == c) break;
  p.nsp += c;
  if (i == str.length) break;
}
```

and then attempt to json decode the rest of the string-form of the packet. At first glance, this should never work, but it turns out for certain values of deadline, this parses perfectly, allowing the client & server modules to communicate. I've been poking around with changing the meaning of things like 'deadline' and when doing so, this bug appeared, which causes the CONNECT packet to be unparseable and dropped, leading to the module halves being unable to communicate with each other.

The fix is to not abuse the namespace in this way, and require a second round-trip to transfer id and rect, OR, allow the namespace to be parsed back to id and rect. The former introduces a new 'pending' state for client connections, which sounds hard, so I'll be preferring the latter approach.
